daemon/endpoint.go

```diff
@@ -436,6 +436,7 @@ func (h *patchEndpointID) Handle(params PatchEndpointIDParams) middleware.Respon
 		}
 	}
 
+	ep.UpdateLogger(nil)
```

Comment:
I feel a bit uneasy that we will forget to call this in places. How do developers know when to call this?

Reply:
@tgraf 
The `UpdateLogger` is used on all Seters that exist for the endpoint and in there, the UpdateLogger is already being called. Would it help if I add a comment where the logger is being created with the fields?

Reply:
Adding a comment to all fields in the endpoint struct to make sure we don't write to them directly is enough for me. Good to see that this is already abstracted into setters.